packages/expo-in-app-purchases/ios/EXInAppPurchases/EXInAppPurchasesModule.m

```diff
@@ -0,0 +1,346 @@
+  NSString *type = [subscriptionPeriod isEqualToString:P0D] ? IN_APP : SUBS;
+
+  NSDecimalNumber *oneMillion = [[NSDecimalNumber alloc] initWithInt:1000000];
+  NSDecimalNumber *priceAmountMicros = [product.price decimalNumberByMultiplyingBy:oneMillion];
+  NSString *price = [NSString stringWithFormat:@"%@%@", product.priceLocale.currencySymbol, product.price];
+
+  return @{
+          @"description": product.localizedDescription,
+          @"price": price,
+          @"priceAmountMicros": priceAmountMicros,
+          @"priceCurrencyCode": product.priceLocale.currencyCode,
+          @"productId": product.productIdentifier,
+          @"subscriptionPeriod": subscriptionPeriod,
+          @"title": product.localizedTitle,
+          @"type": type
+          };
+}
+
+- (NSDictionary *)getTransactionData:(SKPaymentTransaction *)transaction
+{
+  NSData *receiptData = [NSData dataWithContentsOfURL:[[NSBundle mainBundle] appStoreReceiptURL]];
+  return @{
+          @"acknowledged": @YES,
+          @"orderId": transaction.transactionIdentifier,
+          @"productId": transaction.payment.productIdentifier,
+          @"purchaseState": @(transaction.transactionState),
+          @"purchaseTime": @(transaction.transactionDate.timeIntervalSince1970 * 1000),
+          @"transactionReceipt": [receiptData base64EncodedStringWithOptions:0]
+          };
+}
+
+- (NSString *)getSubscriptionPeriod:(SKProduct *)product
+{
+  // Subscription period specified in ISO 8601 format to match Android implementation (e.g. P3M = 3 months)
+  if (@available(iOS 11.2, *)) {
+    NSString *unit = [self getUnit:product];
+    unsigned long numUnits = (unsigned long)product.subscriptionPeriod.numberOfUnits;
+    return [NSString stringWithFormat:@"P%lu%@", numUnits, unit];
+  }
+
+  // Default to P0D if we can't get this info so we assume all products are in app
+  return P0D;
+}
+
+- (NSString *)getUnit:(SKProduct *)product
+{
+  if (@available(iOS 11.2, *)) {
+    switch(product.subscriptionPeriod.unit) {
+      case SKProductPeriodUnitDay: {
+        return @"D";
+      }
+      case SKProductPeriodUnitWeek: {
+        return @"W";
+      }
+      case SKProductPeriodUnitMonth: {
+        return @"M";
+      }
+      case SKProductPeriodUnitYear: {
+        return @"Y";
+      }
+    }
+  }
+  return [NSString string];
+}
+
+- (void)paymentQueue:(SKPaymentQueue *)queue updatedTransactions:(NSArray *)transactions
+{
+  for (SKPaymentTransaction *transaction in transactions) {
+
+    switch(transaction.transactionState) {
+      case SKPaymentTransactionStatePurchasing: {
+        NSLog(@"Transaction state -> Purchasing");
+        break;
+      }
+      case SKPaymentTransactionStatePurchased: {
+        NSLog(@"Made a purchase!");
+        [[SKPaymentQueue defaultQueue] finishTransaction:transaction];
+        NSLog(@"Transaction state -> Purchased");
+        NSArray *results = [NSArray arrayWithObjects: [self getTransactionData:transaction], nil];
```

Comment:
a simpler way to write this is `NSArray *results = @[ [self getTransactionData:transaction], nil ];`